File: SoloLearn/Credit Card Validator/CreditCardValidator.swift
func luhn(cardnumber: String) -> Bool {
    let array = Array(cardnumber)
    var total = 0
    for i in 0...15 {
        if var digit = Int(String(array[i])) {
            if i % 2 == 0 {
                digit = 2 * digit
                if digit > 9 {
                    digit -= 9
                }
            }
            total += digit
        }
    }
    return total % 10 == 0
}

if let cardnumber = readLine() {
    if cardnumber.count == 16 && luhn(cardnumber: cardnumber) {
        print("valid")
    } else {
        print("not valid")
    }
}

File: SoloLearn/Credit Card Validator/CreditCardValidator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoloLearn
{
    class Program
    {
        static bool luhn(string cardnumber) {
            int digit, total = 0;
            for (int i=0; i<16; i++) {
                digit = cardnumber[i] - '0';
                if (i % 2 == 0) {
                    digit = 2 * digit;
                    if (digit > 9)
                        digit -= 9;
                }
                total += digit;
            }
            return total % 10 == 0;
        }

        static void Main(string[] args)
        {
            string cardnumber = Console.ReadLine();
            if (cardnumber.Length == 16 && luhn(cardnumber))
                Console.WriteLine("valid");
            else    
                Console.WriteLine("not valid");
        }
    }
}